### ISSUE TITLE
`dolt pull` fails when other branch exists in the repo

### ISSUE THREAD

**Issue by kretes** [dolthub/dolt#3309]
When doing `dolt pull` I observe a message like this:
```
Updating dugm6punrcogm0jrd7o9uo4fuohmbh67..8grankn555j1c32dgdh3s2nkg5g8iphd
Already up to date.
current fast forward from a to b. a is ahead of b already
```
The problem is that HEAD, main and origin/main are all on `dugm6punrcogm0jrd7o9uo4fuohmbh67`, while `8grankn555j1c32dgdh3s2nkg5g8iphd` is a previous commit. 

Steps to reproduce:
```
mkdir remote_repo && cd remote_repo

dolt init

dolt sql -q "create table a (x int);"
dolt commit -am "table & data added"

dolt checkout -b a_branch
dolt checkout main

dolt sql -q "create table b (x int);"
dolt commit -am "table b in main"

mkdir ../local_repo && cd ../local_repo

dolt clone file:`pwd`/../remote_repo/.dolt/noms .

dolt pull
```

I found that to reproduce it - some other branch is needed.

dolt version: 0.39.2.

This issue isn't happening in 0.39.0, and I suspect this might be related to https://github.com/dolthub/dolt/issues/3113

**Comment by zachmu** [comment]
Thanks for the report, both of you. This is a bug.

We'll get this fixed ASAP.

In the meantime, please either use an earlier release of dolt, or use the separate fetch / merge workflow like:

```
dolt fetch
dolt merge remotes/origin/branch
```

**Comment by timsehn** [comment]
This is likely related to this PR:

https://github.com/dolthub/dolt/pull/3210

@JCOR11599 is on it.

**Comment by JCOR11599** [comment]
Issue should be resolved as of release 0.39.4
